include/stdx/function_traits.hpp

```diff
@@ -29,7 +31,41 @@ template <typename F, template <typename...> typename List>
 using args_t = typename function_traits<F>::template args<List>;
 template <typename F, template <typename...> typename List>
 using decayed_args_t = typename function_traits<F>::template decayed_args<List>;
-template <typename F> using arity_t = typename function_traits<F>::arity;
+template <typename F>
+using nongeneric_arity_t = typename function_traits<F>::arity;
+
+namespace detail {
+template <auto> struct any_type {
+    // NOLINTNEXTLINE(google-explicit-constructor)
+    template <typename T> operator T();
+};
+
+template <typename F, std::size_t... Is>
+constexpr auto try_invoke_impl(std::index_sequence<Is...>)
+    -> std::invoke_result_t<F, any_type<Is>...>;
+
+template <typename F, typename N>
+using try_invoke =
+    decltype(try_invoke_impl<F>(std::make_index_sequence<N::value>{}));
 
+template <typename F, typename N>
+using has_arg_count = boost::mp11::mp_valid<try_invoke, F, N>;
+
+template <typename F, typename N> struct generic_arity;
+
+template <typename F, typename N>
+using generic_arity_t = typename generic_arity<F, N>::type;
+
+template <typename F, typename N> struct generic_arity {
+    using type = boost::mp11::mp_eval_if<
+        has_arg_count<F, N>, N, generic_arity_t, F,
+        std::integral_constant<std::size_t, N::value + 1u>>;
+};
+} // namespace detail
+
+template <typename F>
+using arity_t = boost::mp11::mp_eval_or<
```

Comment:
suggestion: would be nice to have a corresponding `arity_v`

Reply:
:+1: added